File: 4264-FirstMatchingCharacterFromBothEnds/4264-FirstMatchingCharacterFromBothEnds.java
// Last updated: 3/29/2026, 11:14:44 AM
class Solution {
    public int firstMatchingIndex(String s) {
        char[] arr = s.toCharArray();
        int n = s.length();
        for(int index = 0 ; index < n ; index++){
            if(arr[index]==arr[n-index-1]) return index;
        }
        return -1;
    }
}

File: 4264-FirstMatchingCharacterFromBothEnds/4264-FirstMatchingCharacterFromBothEnds.cpp
// Last updated: 3/29/2026, 11:14:46 AM
class Solution {
public:
    int firstMatchingIndex(string s) {
        for(int index = 0 ; index < s.size() ; index++){
            if(s[index]==s[s.size()-index-1]) return index;
        }
        return -1;
    }
};